File: 01_Language/01_Functions/js/chunk_split.js
function chunk_split(body, chunklen = 76, end = "\r\n") {
    let res = ""
    let str = ""
    let start = 0;
    while ((str = body.substring(start, start + chunklen)) !== "") {
        res += str + end
        start += chunklen
    }
    return res;
}

const base64_encode = require("./base64_encode")
// format $data using RFC 2045 semantics
let data = "Returns the chunked string.";
let new_string = chunk_split(base64_encode(data) + base64_encode(data) + base64_encode(data) + base64_encode(data) +base64_encode(data));
console.log(new_string);

File: 01_Language/01_Functions/python/chunk_split.py
# coding: utf-8

import textwrap


def chunk_split(body, chunklen=76, end="\r\n"):
    return end.join(textwrap.wrap(body, chunklen))


if __name__ == '__main__':
    from base64_encode import base64_encode

    # format $data using RFC 2045 semantics
    data = "Returns the chunked string."
    print(repr(chunk_split(base64_encode(data) * 5)))
